pkg/codegen/schema/schema.go

```diff
@@ -1218,13 +1267,14 @@ func (t *types) bindEnumType(token string, spec ComplexTypeSpec) (*EnumType, err
 
 func bindTypes(pkg *Package, complexTypes map[string]ComplexTypeSpec) (*types, error) {
 	typs := &types{
-		pkg:     pkg,
-		objects: map[string]*ObjectType{},
-		arrays:  map[Type]*ArrayType{},
-		maps:    map[Type]*MapType{},
-		unions:  map[string]*UnionType{},
-		tokens:  map[string]*TokenType{},
-		enums:   map[string]*EnumType{},
+		pkg:       pkg,
+		resources: map[string]*ResourceType{},
+		objects:   map[string]*ObjectType{},
+		arrays:    map[Type]*ArrayType{},
+		maps:      map[Type]*MapType{},
+		unions:    map[string]*UnionType{},
+		tokens:    map[string]*TokenType{},
+		enums:     map[string]*EnumType{},
 	}
 
 	// Declare object and enum types before processing properties.
```

Comment:
You should add a pass to declare resource types here.

Reply:
I had moved this to `bindResource` because not every `'object'` is a resource. I've moved it back and added a conditional that checks for `#/resource` in the token before adding it to the resource map.